src/Kit/System/Watchdog/_0test/_rawthread/rawthread.cpp

```diff
@@ -0,0 +1,335 @@
+
+    // Check if this is a watchdog reset
+    if ( __HAL_RCC_GET_FLAG( RCC_FLAG_IWDGRST ) != RESET )
+    {
+        const char* successMsg = "*** WATCHDOG RESET DETECTED ***\r\n";
+        HAL_UART_Transmit(&huart3, (uint8_t*)successMsg, strlen(successMsg), 1000);
+        const char* successMsg2 = "Test PASSED - System was reset by watchdog when raw thread timed out\r\n";
+        HAL_UART_Transmit(&huart3, (uint8_t*)successMsg2, strlen(successMsg2), 1000);
+        
+        __HAL_RCC_CLEAR_RESET_FLAGS();
+
+        // Keep both LEDs on to indicate success
+        Bsp_turn_on_debug1();
+        Bsp_turn_on_debug2();
+
+        // Infinite loop - test is complete
+        while ( 1 )
+        {
+            HAL_Delay( 1000 );
+        }
+    }
+
+    const char* msg2 = "KIT System initialized\r\n";
+    HAL_UART_Transmit(&huart3, (uint8_t*)msg2, strlen(msg2), 1000);
+
+    char buf[200];
+    snprintf(buf, sizeof(buf), "Raw Thread Stuck Test Starting\r\nHW Watchdog Timeout: %u ms\r\nRaw Thread Timeout: %u ms\r\nStuck Duration: %u ms\r\n", 
+             HW_WATCHDOG_TIMEOUT_MS, RAW_THREAD_WDOG_TIMEOUT_MS, STUCK_DURATION_MS);
+    HAL_UART_Transmit(&huart3, (uint8_t*)buf, strlen(buf), 1000);
+
+    // First run - set up the test
+    const char* msg3 = "First run - setting up watchdog test\r\n";
+    HAL_UART_Transmit(&huart3, (uint8_t*)msg3, strlen(msg3), 1000);
+
+    // Configure the supervisor thread (event thread) using placement new
+    WatchedEventThread* supervisorWdog = new (supervisorWdogStorage_) WatchedEventThread(
+        SUPERVISOR_WDOG_TIMEOUT_MS,
+        SUPERVISOR_HEALTH_CHECK_MS,
+        true );  // This is the supervisor thread
+
+    // Create the event loop for the supervisor thread using placement new
+    Server* eventLoop = new (eventLoopStorage_) Server(
+        OPTION_KIT_SYSTEM_EVENT_LOOP_TIMEOUT_PERIOD,
+        nullptr,
+        supervisorWdog );
+
+    // Create and start the supervisor thread
+    auto* supervisorThread = Thread::create( *eventLoop, "SUPERVISOR" );
+    if ( !supervisorThread )
+    {
+        FatalError::logf( Shutdown::eFAILURE, "Failed to create supervisor thread" );
+    }
+    const char* msg4 = "Supervisor thread created\r\n";
+    HAL_UART_Transmit(&huart3, (uint8_t*)msg4, strlen(msg4), 1000);
+
+    // Create and start the raw thread using placement new
+    RawThreadRunnable* rawRunnable = new (rawRunnableStorage_) RawThreadRunnable();
+    auto* rawThread = Thread::create( *rawRunnable, "RAW_THREAD" );
+    if ( !rawThread )
+    {
+        FatalError::logf( Shutdown::eFAILURE, "Failed to create raw thread" );
+    }
+    const char* msg5 = "Raw thread created\r\n";
+    HAL_UART_Transmit(&huart3, (uint8_t*)msg5, strlen(msg5), 1000);
+
+    // Create and start the test monitor thread using placement new
+    TestMonitor* testMonitor = new (testMonitorStorage_) TestMonitor();
+    auto* monitorThread = Thread::create( *testMonitor, "TEST_MONITOR" );
+    if ( !monitorThread )
```

Comment:
Does not need to be dynamically allocated.
```suggestion
    static TestMonitor testMonitor_;
    auto* monitorThread = Thread::create( testMonitor_, "TEST_MONITOR" );
    if ( !monitorThread )
```